### PR TITLE
Check that the ceph-csi version matches the OCP version on the hosted clusters

### PR DESCRIPTION
**Pull request by yitzhak12**
Check that the ceph-csi version matches the OCP version on the hosted clusters.
The test will perform the following:
  1. Iterate over the clients
  2. Get the ceph csi ctrl pods container images of the client
  3. Get the csi images for the client OCP version
  4. Check that the ceph csi ctr pod container images exist in the csi images of the client OCP version

### CODE REVIEW

**Comment by yitzhak12** [comment]
The PR validation job passed here: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/2693/.

File: tests/functional/provider_mode/test_ceph_csi_image_versions.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        Switch to the original cluster index
+        """
+
+        def finalizer():
+            log.info("Switch to the original cluster index")
+            config.switch_ctx(self.orig_index)
+
+        request.addfinalizer(finalizer)
+
+    def test_client_clusters_csi_image_versions(self):
```
Comment (DanielOsypenko) [inline tests/functional/provider_mode/test_ceph_csi_image_versions.py:43]: we'll need polarion id here

**Review by jilju — DISMISSED** [review]

File: tests/functional/provider_mode/test_ceph_csi_image_versions.py (hunk 1 above)
Comment (yitzhak12) [reply tests/functional/provider_mode/test_ceph_csi_image_versions.py:43]: Ack

Comment (yitzhak12) [reply tests/functional/provider_mode/test_ceph_csi_image_versions.py:44]: Fixed.

**Comment by yitzhak12** [comment]
The PR validation job passed: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/2729/

**Review by fbalak — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Pull request merged**